include/tts/tests/range.hpp

```diff
@@ -218,8 +221,8 @@ namespace tts
           {
             histogram[i]       += local_histogram[i];
             sample_values[i]    = std::min(sample_values[i], local_sample_values[i]);
-            expected_values[i]  = *::tts::detail::begin(f(sample_values[i]));
-            result_values[i]    = *::tts::detail::begin(g(sample_values[i]));
+            expected_values[i]  = *::tts::detail::begin(f(T(sample_values[i])));
```

Comment:
why the cast ? This will probably break SIMD tests

Reply:
non, au contraire j'avais une fonction qui etait ok en scalaire et pas en simd et sans le T ça me disais x au lie de x